commands/outdated.js

```diff
@@ -125,21 +137,38 @@ const outdated /*: Outdated */ = async ({root, logger = console.log}) => {
 
   for (const name in info) {
     const latest = info[name].version;
-    if (latest && typeof latest === 'string') {
-      for (const range of map[name]) {
-        if (!validRange(range) || !validRange(latest)) {
-          continue;
-        }
-        if (gt(latest, minVersion(range))) {
-          results.push({name, range: [range], latest});
+    if (latest && typeof latest === 'string' && validRange(latest)) {
+      const outdated = [];
+      for (const version of map[name]) {
+        if (validRange(version) && gt(latest, minVersion(version))) {
+          outdated.push(version);
         }
       }
+      if (outdated.length > 0) {
+        results.push({packageName: name, installed: outdated, latest});
+      }
     }
   }
 
   // report discrepancies
-  for (const {name, range, latest} of results) {
-    logger(name, range[0], latest);
+  for (const result of results) {
+    const formatted = [];
+    if (dedup) {
+      formatted.push(result);
+    } else {
+      result.installed.forEach(version =>
+        formatted.push({
+          ...result,
+          installed: [version],
+        })
+      );
+    }
+
+    formatted.forEach(entry =>
+      json
+        ? logger(JSON.stringify(entry))
```

Comment:
Technically only each entry is valid JSON.  In aggregate, the output isn't valid.  If we have some strong opinions here against this convention, happy to think through alternatives.

Reply:
Any reason not to wrap an array around it? Would allow piping to jq

Reply:
That does mean we aren't only passing entries to the provided `logger`.  So folks that do provide a custom logger would need to special case the first/last call.

I think that is probably a fine trade-off since the primary use cases are `console.log` and unit tests 🤷

Reply:
Updated to ensure `--json` output is valid JSON